File: internal/auth/auth_test.go
package auth

import (
	"net/http"
	"testing"
)

func TestGetAPIKey_Valid(t *testing.T) {

	headers := http.Header{}
	headers.Set("Authorization", "ApiKey my-secret-key")

	got := GetAPIKey(headers)
	if got != "my-secret-key" {
		t.Errorf("expected %q, got %q", "my-secret-key", got)
	}
}

func TestGetAPIKey_InvalidPrefix(t *testing.T) {
	headers := http.Header{}
	headers.Set("Authorization", "Bearer something")

	got := GetAPIKey(headers)
	if got != "" {
		t.Errorf("expected empty string, got %q", got)
	}
}

func TestGetAPIKey_Missing(t *testing.T) {
	headers := http.Header{}

	got := GetAPIKey(headers)
	if got != "" {
		t.Errorf("expected empty string, got %q", got)
	}
}

File: src/tests/api/auth.test.ts
import { describe, expect, test } from "vitest";
import { getAPIKey } from "../../api/auth.js";
import type { IncomingHttpHeaders } from "http";

function createHeaders(
  overrides?: Partial<IncomingHttpHeaders>,
): IncomingHttpHeaders {
  return { ...overrides } as IncomingHttpHeaders;
}

describe("getAPIKey", () => {
  test("returns null when authorization header is missing", () => {
    const headers = createHeaders();
    expect(getAPIKey(headers)).toBeNull();
  });

  test("returns null when authorization header is empty string", () => {
    const headers = createHeaders({ authorization: "" });
    expect(getAPIKey(headers)).toBeNull();
  });

  test("returns null when authorization uses wrong scheme (Bearer instead of ApiKey)", () => {
    const headers = createHeaders({ authorization: "Bearer some-token" });
    expect(getAPIKey(headers)).toBeNull();
  });

  test("returns null when authorization has ApiKey but no key value", () => {
    const headers = createHeaders({ authorization: "ApiKey" });
    expect(getAPIKey(headers)).toBeNull();
  });

  test("returns null when authorization has extra spaces but wrong scheme", () => {
    const headers = createHeaders({ authorization: "  Bearer token  " });
    expect(getAPIKey(headers)).toBeNull();
  });

  test("returns API key when format is ApiKey <key>", () => {
    const headers = createHeaders({
      authorization: "ApiKey my-secret-key-123",
    });
    expect(getAPIKey(headers)).toBe("wrong-value-to-fail");
  });

  test("returns key when key is empty string (ApiKey followed by space)", () => {
    const headers = createHeaders({ authorization: "ApiKey " });
    expect(getAPIKey(headers)).toBe("");
  });
});
